File: handoffs/a2-scripts/probit_tier1.py
import warnings, numpy as np
from pystatistics.gam import gam, s
from pystatistics.regression.families import Binomial
d = np.genfromtxt("fs_binomial.csv", delimiter=",", names=True)
y = d["y"].astype(float)
with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    sol = gam(y, smooths=[s("x1",k=10,bs="cr"), s("x2",k=8,bs="cr")],
              smooth_data={"x1": d["x1"], "x2": d["x2"]},
              family=Binomial(link="probit"), method="REML",
              sp=[40.00448, 23.14561])
fit_r = np.genfromtxt("probit_t1_fitted.csv", delimiter=",", skip_header=1)
print(f"py probit @fixed sp: edf={sol.total_edf:.8f} dev={sol.deviance:.8f} "
      f"reml={sol.reml_score:.8f}")
print("fitted max abs diff vs mgcv:", float(np.max(np.abs(sol.fitted_values - fit_r))))

File: handoffs/a2-scripts/probit_tier1.R
suppressMessages(library(mgcv))
d <- read.csv("fs_binomial.csv")
# fixed at mgcv's own REML-selected sp
m <- gam(y ~ s(x1,k=10,bs="cr") + s(x2,k=8,bs="cr"), family=binomial(link=probit),
         method="REML", sp=c(40.00448,23.14561), data=d)
cat(sprintf("mgcv probit @fixed sp: edf=%.8f dev=%.8f reml=%.8f\n", sum(m$edf), deviance(m), m$gcv.ubre))
write.csv(data.frame(fitted=fitted(m)), "probit_t1_fitted.csv", row.names=FALSE)
